lib/chef/knife.rb

```diff
@@ -26,14 +26,24 @@
 require 'chef/knife/core/subcommand_loader'
 require 'chef/knife/core/ui'
 require 'chef/local_mode'
-require 'chef/rest'
+require 'chef/server_api'
 require 'chef/http/authenticator'
+require 'chef/http/http_request'
+require 'chef/http'
 require 'pp'
 
 class Chef
   class Knife
 
-    Chef::REST::RESTRequest.user_agent = "Chef Knife#{Chef::REST::RESTRequest::UA_COMMON}"
+    class NoAuthHTTP < Chef::HTTP
+      use Chef::HTTP::JSONInput
+      use Chef::HTTP::JSONOutput
+      use Chef::HTTP::CookieManager
+      use Chef::HTTP::Decompressor
+      use Chef::HTTP::RemoteRequestID
```

Comment:
`use ValidateContentLength` should be added here (after Decompressor)

probably should move this to Chef::HTTP::JSONSimple / Chef::HTTP::Simple::JSON or something for re-use..